compiler/src/iree/compiler/Codegen/Utils/GPUUtils.cpp

```diff
@@ -438,6 +438,17 @@ static Value warpReduction(Location loc, OpBuilder &builder, Value input,
   const bool needsPacking = kShuffleBitWidth != origBitWidth;
   IntegerType equivIntType = builder.getIntegerType(origBitWidth);
 
+  // Defer expansion of subgroup reduction until later in pass pipeline to
+  // enable conditional lowering to DPP ops, for potential perf gains over
+  // gpu.shuffle ops.
+
+  // FOR REVIEWERS: should this be a numLaneToReduce <= warpSize,
+  // and then pass numLaneToReduce as cluster size to the subgroup_reduce op?
+  if (numLaneToReduce == warpSize) {
```

Comment:
Yes, this should be a <= - but also with `warpSize % numLaneToReduce == 0` - and that should be cluster size